### ISSUE TITLE
SafeAreaView not working on Android. Giving wrong paddings

### ISSUE THREAD

**Issue by rogerkerse** [facebook/react-native#17371]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

Environment:
  OS: macOS High Sierra 10.13.2
  Node: 9.3.0
  Yarn: 1.3.2
  npm: 5.5.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  react: 16.0.0 => 16.0.0
  react-native: 0.51.0 => 0.51.0

Target Platform: Android (8.1.0) Pixel 2 XL

### Steps to Reproduce

1. Add view setup like this
```
<SafeAreaView style={{ flex: 1 }}>
    <View style={{ flex: 1 }} onLayout={this.onLayout} />
</SafeAreaView>
```
2. Add onLayout function like this
```
onLayout = (event) => {
    const { y: top, height } = event.nativeEvent.layout
    const { height: screenHeight } = Dimensions.get('window')

    const safeAreaBottomPadding = screenHeight - height - top

    console.log(top, safeAreaBottomPadding) // prints 0, 27.9999.
}
```

### Expected Behavior

Bottom padding should be calculated 0.

### Actual Behavior

The bottom padding comes out 27.9999 which is half the size of bottom soft navigation bar so it does not make sense. (Top status bar seems about 28 pixels, but y gets value of 0).

### Reproducible Demo

(Paste the link to an example project and exact instructions to reproduce the issue.)

**Comment by waquidvp** [comment]
The way SafeAreaView is set up, on Android it shouldn't have any affect. SafeAreaView only has affect on iOS, specifically on the iPhone X.

**Comment by rogerkerse** [comment]
@waquidvp  But if you try it out it gives bottom padding, which has an effect

**Comment by burkeshartsis** [comment]
Currently `SafeAreaView` on android just returns a `View`.

https://github.com/facebook/react-native/blob/master/Libraries/Components/SafeAreaView/SafeAreaView.android.js

**Comment by krmao** [comment]
SafeAreaView for android just return view may be not a good idea,
I think it should be return StatusBar.currentHeight .
and for ios return safe statusbar height

**Comment by hramos** [comment]
It does not look like this has been reproduced in 0.54 or later. There has been no follow up from the author in over seven days, so we're closing this issue.

Furthermore, SAV is just a View on Android.

**Comment by Thomas-Calliweb** [comment]
Hi, i've got this kind of issue, i'm testing my app on my android phone and it has a notch (one plus 6) at the top.

SafeAreaView will be usefull to be effective on android too, and if it is not possible, what is the solution to prevent absolute element to be under the topbar.

Thanks.

**Comment by szemannp** [comment]
Indeed, this issue becomes more relevant as the notched Android devices are spreading - OP6, Huawei Pro 20, LG G7 etc.

**Comment by amerllica** [comment]
In the `SafeAreaView` [Docs](https://facebook.github.io/react-native/docs/safeareaview) was told:

> It is currently only applicable to iOS devices with iOS version 11 or later.

So now I definitely use it in my project but I use `Platform` to recognize devide platfrom and for Android I make a manual safe area for status bar.

**Comment by pribeh** [comment]
@amalChandran I'm doing the same.

**Comment by sgtpepper43** [comment]
There's a new api for android devices called [DisplayCutout](https://developer.android.com/reference/android/view/DisplayCutout) that can be used for dealing with notches. We should consider using this with `SafeAreaView`, since there are quite a few android devices with notches coming out.

**Comment by morenoh149** [comment]
@sgtpepper43 start an issue as a discussion, see below 
<img width="781" alt="screen shot 2018-12-22 at 1 09 21 am" src="https://user-images.githubusercontent.com/587438/50371322-4e5d3a80-0586-11e9-85c0-42bb9443f5ab.png">

**Comment by yodaheis** [comment]
Can we use the difference between Dimensions.get('screen') and Dimensions.get('window') to calculate status bar height and set safe margin according to the orientation?

**Comment by AhmedMKamal** [comment]
@yodaheis  `Dimensions.get('screen')` returns the `width/height` of the entire screen without status bar, soft menu bar or navigation bar while `Dimensions.get('window')`returns the `width/height` of the application view "screen size without status bar, soft menu bar or navigation bar while" but when it comes to notches I'm not sure but your suggestion makes sense for me I may give it a try.

**Comment by sgtpepper43** [comment]
There's already an API to get notches on Android, we don't need to do any weird hacks to do it.

**Comment by AhmedMKamal** [comment]
@sgtpepper43 notches API is available only in Android 9.0!! please tell me if I'm wrong?

**Comment by AhmedMKamal** [comment]
**TL;DR**
I tested this out and what I ended with is that the notch in Android is just a part of the status bar so dealing with `StatusBar.currentHeight` makes your life easier when dealing with notches, `StatusBar.currentHeight` on a normal device returns `24` while on a notched device it returns `48` keep in mind this is the behavior I got on a simulated devices running Android 9.0 so I'm not sure if it's the default behavior.
**But**, `SafeAreaView` adds some extra padding on Android so I ended up with something like that 
```
<SafeAreaView 
    style={styles.container}
    forceInset={Platform.OS === 'android' && { vertical: 'never' }}>
        <StatusBar />{/* this is a custom StatusBar component */}
        <AppNavigator />
</SafeAreaView>
```
this gave me the following results on Android

![screenshot-2019-02-18_22 42 57 91](https://user-images.githubusercontent.com/35304348/52976663-fea6be00-33d2-11e9-8e7a-81e9fdda2da6.png)

=====================================================================

![screenshot-2019-02-18_22 39 18 616](https://user-images.githubusercontent.com/35304348/52976670-01a1ae80-33d3-11e9-86a4-b27c113678aa.png)

Hope that makes sense ¯\_(ツ)_/¯

**Comment by sgtpepper43** [comment]
@AhmedMKamal yeah, the API is only available on Android 9, but almost all phones with notches are brand new and running Android 9. There are a most a small handful of phones you'd have to worry about that don't support the API and also have notches. Like literally three phones probably

**Comment by sbatezat** [comment]
@sgtpepper43 Three is still too much...
I've got the issue on a Huawei p20 pro which has not installed the update to Android 9. We need to think about these users too.

**Comment by sgtpepper43** [comment]
Sure we need to think about these users, but by the time React Native gets updated with _any_ fixes how many notched phones will still be on Android 8?

**Comment by sbatezat** [comment]
@sgtpepper43 Probably not a lot, but still some of them.
For the time being, I'm hardcoding a paddingTop on containers (see https://stackoverflow.com/a/51427630)
